API docs documentation

miguelangel-dev opened the pull request:
## Description
Complete API docs for `nef`. It reaches the goal of 100% code coverage.

### Documented modules:
- `nef`
- `NefModels`

![Captura de pantalla 2020-03-23 a las 14 19 20](https://user-images.githubusercontent.com/25568562/77320679-4ce26b00-6d11-11ea-87d5-330c965e7b69.png)

miguelangel-dev committed:
complete api-docs for carbon models

miguelangel-dev committed:
api docs for Carbon View

miguelangel-dev committed:
api docs for the platforms definitions

miguelangel-dev committed:
api docs for PlaygroundExcludeItem

miguelangel-dev committed:
api docs for PlaygroundDependencies

miguelangel-dev committed:
complete the api docs for nef APIs

miguelangel-dev committed:
minor changes

miguelangel-dev committed:
api docs for each `case` in NefModels

miguelangel-dev committed:
api docs for kleisli help to transform an EnvIO into a IO using default console

miguelangel-dev committed:
api docs for nef Error

miguelangel-dev committed:
api docs for Console

Update project/Component/NefModels/CarbonModel.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/CarbonModel.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/CarbonModel.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/CarbonModel.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/CarbonModel.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/Console.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/nef/Utils/Kleisli+API.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/nef/Utils/Kleisli+API.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/nef/Utils/Kleisli+API.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/CarbonModel.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/nef/Utils/Kleisli+API.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/CarbonModel.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/CarbonModel.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/CarbonModel.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/CarbonModel.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/CarbonModel.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/CarbonModel.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/CarbonModel.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/Console.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/CarbonModel.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/CarbonModel.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/CarbonModel.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/CarbonModel.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/CarbonModel.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/CarbonModel.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/CarbonModel.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

truizlop commented on project/Component/NefModels/CarbonModel.swift:
```diff
@@ -2,25 +2,54 @@
 
+/// Represents the configuration for rendering carbon snippets.
 public struct CarbonModel: Codable, Equatable {
+    /// Code to render into Carbon image.
```
```suggestion
    /// Code to render as a Carbon image.
```

truizlop commented on project/Component/NefModels/CarbonModel.swift:
```diff
@@ -2,25 +2,54 @@
+    /// Initializes a `CarbonModel`
+    ///
+    /// - Parameters:
+    ///   - code: code to render into Carbon image.
```
```suggestion
    ///   - code: code to render as a Carbon image.
```

truizlop commented on project/Component/NefModels/CarbonModel.swift:
```diff
@@ -2,25 +2,54 @@
+    /// Background color
     public let background: Color
+    
+    /// File size.
```
I think *file size* is misguiding.
```suggestion
    /// Image size.
```

truizlop commented on project/Component/NefModels/CarbonModel.swift:
```diff
@@ -2,25 +2,54 @@
+    /// - Parameters:
+    ///   - background: background color in hexadecimal.
+    ///   - theme: carbon theme.
+    ///   - size: export file size.
```
```suggestion
    ///   - size: exported image size.
```

truizlop commented on project/Component/NefModels/CarbonModel.swift:
```diff
@@ -2,25 +2,54 @@
+    /// Initializes a `CarbonStyle`
+    ///
+    /// - Parameters:
+    ///   - background: background color in hexadecimal.
```
The hexadecimal detail should be hidden in the implementation of `Color`.

```suggestion
    ///   - background: background color.
```

truizlop commented on project/Component/NefModels/CarbonModel.swift:
```diff
@@ -2,25 +2,54 @@
+    ///
+    /// - Parameters:
+    ///   - background: background color in hexadecimal.
+    ///   - theme: carbon theme.
```
```suggestion
    ///   - theme: Carbon theme.
```

truizlop commented on project/Component/NefModels/CarbonModel.swift:
```diff
@@ -2,25 +2,54 @@
+    ///   - background: background color in hexadecimal.
+    ///   - theme: carbon theme.
+    ///   - size: export file size.
+    ///   - fontType: carbon font type.
```
```suggestion
    ///   - fontType: Carbon font type.
```

truizlop commented on project/Component/NefModels/CarbonModel.swift:
```diff
@@ -30,70 +59,166 @@ public struct CarbonStyle: Codable, Equatable {
         self.watermark = watermark
     }
     
+    /// Represents file size.
```
```suggestion
    /// Represents image size.
```

truizlop commented on project/Component/NefModels/CarbonModel.swift:
```diff
@@ -30,70 +59,166 @@ public struct CarbonStyle: Codable, Equatable {
     
+    /// Represents file size.
     public enum Size: Double, Codable, Equatable, CaseIterable {
+        /// Output file size to 1x
```
```suggestion
        /// Output image size to 1x
```

truizlop commented on project/Component/NefModels/CarbonModel.swift:
```diff
@@ -30,70 +59,166 @@ public struct CarbonStyle: Codable, Equatable {
     public enum Size: Double, Codable, Equatable, CaseIterable {
+        /// Output file size to 1x
         case x1 = 1
+        /// Output file size to 2x
```
```suggestion
        /// Output image size to 2x
```

truizlop commented on project/Component/NefModels/CarbonModel.swift:
```diff
@@ -30,70 +59,166 @@ public struct CarbonStyle: Codable, Equatable {
         case x1 = 1
+        /// Output file size to 2x
         case x2 = 2
+        /// Output file size to 3x
```
```suggestion
        /// Output image size to 3x
```

truizlop commented on project/Component/NefModels/CarbonModel.swift:
```diff
@@ -30,70 +59,166 @@ public struct CarbonStyle: Codable, Equatable {
         case x2 = 2
+        /// Output file size to 3x
         case x3 = 3
+        /// Output file size to 4x
```
```suggestion
        /// Output image size to 4x
```

truizlop commented on project/Component/NefModels/CarbonModel.swift:
```diff
@@ -30,70 +59,166 @@ public struct CarbonStyle: Codable, Equatable {
         case x3 = 3
+        /// Output file size to 4x
         case x4 = 4
+        /// Output file size to 5x
```
```suggestion
        /// Output image size to 5x
```

truizlop commented on project/Component/NefModels/CarbonModel.swift:
```diff
@@ -30,70 +59,166 @@ public struct CarbonStyle: Codable, Equatable {
         case x5 = 5
     }
     
+    /// Represents Carbon theme.
```
```suggestion
    /// Represents a Carbon theme.
```

truizlop commented on project/Component/NefModels/CarbonModel.swift:
```diff
@@ -30,70 +59,166 @@ public struct CarbonStyle: Codable, Equatable {
+        /// Base 16 dark
         case base16 = "base16-dark"
+        
+        /// Balckboard
```
```suggestion
        /// Blackboard
```

truizlop commented on project/Component/NefModels/CarbonModel.swift:
```diff
@@ -30,70 +59,166 @@ public struct CarbonStyle: Codable, Equatable {
         case zenburn
     }
     
+    /// Represents Carbon font type
```
```suggestion
    /// Represents a Carbon font type
```

truizlop commented on project/Component/NefModels/CarbonModel.swift:
```diff
@@ -124,30 +253,56 @@ extension CarbonStyle.Color {
+    /// The snippet of code fails in the action.
     public let snippet: String
+    
+    /// Describe the reason for the failure.
```
```suggestion
    /// Describes the reason for the failure.
```

truizlop commented on project/Component/NefModels/CarbonModel.swift:
```diff
@@ -124,30 +253,56 @@ extension CarbonStyle.Color {
+/// Represents an error in a Carbon action.
 public struct CarbonError: Error {
+    
+    /// The snippet of code fails in the action.
```
```suggestion
    /// Snippet that was being rendered when the action failed.
```

truizlop commented on project/Component/NefModels/CarbonModel.swift:
```diff
@@ -124,30 +253,56 @@ extension CarbonStyle.Color {
+    /// Initializes a `CarbonError`
+    ///
+    /// - Parameters:
+    ///   - snippet: the snippet of code fails in the action
```
```suggestion
    ///   - snippet: snippet of code that was being rendered when the action failed.
```

truizlop commented on project/Component/NefModels/CarbonModel.swift:
```diff
@@ -124,30 +253,56 @@ extension CarbonStyle.Color {
+    ///
+    /// - Parameters:
+    ///   - snippet: the snippet of code fails in the action
+    ///   - cause: the reason for the failure.
```
```suggestion
    ///   - cause: reason for the failure.
```

truizlop commented on project/Component/NefModels/CarbonModel.swift:
```diff
@@ -124,30 +253,56 @@ extension CarbonStyle.Color {
+        /// Could not take a snapshot.
         case invalidSnapshot
         
+        /// A textual representation of `CarbonError.Cause`.
```
I think this does not need to be documented, as its documentation is inherited from `CustomStringConvertible`.

miguelangel-dev replied:
yep, you are right - I think so, anyway, I will check it

truizlop commented on project/Component/NefModels/CarbonModel.swift:
```diff
@@ -157,8 +312,11 @@ public struct CarbonError: Error {
+    
+    /// Initializes a `CarbonStyle.Size`
+    ///
+    /// - Parameter factor: export file size.
```
```suggestion
    /// - Parameter factor: export image size.
```

truizlop commented on project/Component/NefModels/CarbonModel.swift:
```diff
@@ -167,6 +325,10 @@ extension CarbonStyle.Size {
+    
+    /// Initializes a `CarbonStyle.Color`
+    ///
+    /// - Parameter hex: creates a `Color` from an hexadecimal.
```
```suggestion
    /// - Parameter hex: creates a `Color` from a hexadecimal String.
```

truizlop commented on project/Component/NefModels/Console.swift:
```diff
@@ -16,25 +28,59 @@ public struct Step {
+    /// Defined dummy `Step`
     public static var empty: Step { .init(total: 0, partial: 0, duration: .never) }
     
+    /// Advance i-operations the step.
```
```suggestion
    /// Advance the step a number of increments.
```

truizlop commented on project/Component/NefModels/Console.swift:
```diff
@@ -16,25 +28,59 @@ public struct Step {
     public static var empty: Step { .init(total: 0, partial: 0, duration: .never) }
     
+    /// Advance i-operations the step.
+    /// - Parameter partial: number of i-operations.
```
```suggestion
    /// - Parameter partial: number of increments for the step.
```

truizlop commented on project/Component/nef/Utils/Kleisli+API.swift:
```diff
@@ -5,16 +5,30 @@ import NefModels
+/// Transforms an `EnvIO<Console, nef.Error, A>` into `IO<nef.Error, A>` using a dummy console.
 public extension Kleisli where D == NefModels.Console, F == IOPartial<nef.Error> {
+    
+    /// Performs the side effects that are suspended in this IO in a synchronous manner. Resolve `D` dependency of type `Console` using a dummy console.
```
```suggestion
    /// Performs the side effects that are suspended in this IO in a synchronous manner, resolving the dependency of type `Console` using a dummy implementation.
```

truizlop commented on project/Component/nef/Utils/Kleisli+API.swift:
```diff
@@ -5,16 +5,30 @@ import NefModels
         try provide(MacDummyConsole()).unsafeRunSync()
     }
     
+    /// Performs the side effects that are suspended in this IO in a synchronous manner. Resolve `D` dependency of type `Console` using a dummy console.
```
```suggestion
    /// Performs the side effects that are suspended in this IO in a synchronous manner, resolving the dependency of type `Console` using a dummy implementation.
```

truizlop commented on project/Component/nef/Utils/Kleisli+API.swift:
```diff
@@ -5,16 +5,30 @@ import NefModels
     }
     
-    func unsafeRunAsync(with d: D, on queue: DispatchQueue = .main, _ callback: @escaping Callback<nef.Error, A>) {
+    /// Performs the side effects that are suspended in this IO in an asynchronous manner. Resolve `D` dependency of type `Console` using a dummy console.
```
```suggestion
    /// Performs the side effects that are suspended in this IO in an asynchronous manner, resolving the dependency of type `Console` using a dummy implementation.
```

truizlop commented on project/Component/nef/Utils/Kleisli+API.swift:
```diff
@@ -5,16 +5,30 @@ import NefModels
+    /// - Parameters
+    ///   - queue: Dispatch queue used to execute the side effects. Defaults to the main queue.
+    ///   - callback: A callback function to receive the results of the evaluation.
+    /// - Returns: Value produced after running the suspended side effects.
```
It does not return anything.

```suggestion
```

truizlop commented:
A general comment: the description of each parameter should start with a capital letter, as it is the beginning of a sentence. Use that as a convention.

miguelangel-dev committed:
comments in PR

minor changes

minor changes

Update project/Component/NefModels/CarbonView.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/CarbonView.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/CarbonModel.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/CarbonModel.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/CarbonModel.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/nef/Models/Error.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

comments in PR

truizlop commented on project/Component/NefModels/CarbonModel.swift:
```diff
@@ -2,25 +2,54 @@
+    ///   - size: Exported image size.
+    ///   - fontType: Carbon font type.
+    ///   - lineNumbers: Shows/hides lines of code.
+    ///   - watermark:Shows/hides the watermark.
```
```suggestion
    ///   - watermark: Shows/hides the watermark.
```

truizlop commented on project/Component/NefModels/CarbonModel.swift:
```diff
@@ -30,70 +59,166 @@ public struct CarbonStyle: Codable, Equatable {
+        /// Material
         case material
+        
+        ///Monokai
```
```suggestion
        /// Monokai
```

truizlop commented on project/Component/NefModels/CarbonModel.swift:
```diff
@@ -114,8 +240,10 @@ extension CarbonStyle: CustomStringConvertible {
+/// Predefined Carbon `Color`.
 extension CarbonStyle.Color {
+    
+    /// A group of predefined Carbon colours.
```
```suggestion
    /// A group of predefined Carbon colors.
```

truizlop commented on project/Component/NefModels/CarbonView.swift:
```diff
@@ -2,12 +2,22 @@
+/// Describes a Carbon loading view.
 public protocol CarbonLoadingView: NSView {
+    
+    /// Loading view should change its state to show.
```
```suggestion
    /// Shows the loading view.
```

truizlop commented on project/Component/NefModels/CarbonView.swift:
```diff
@@ -2,12 +2,22 @@
+    /// Loading view should change its state to show.
     func show()
+    
+    /// Loading view should change its state to hidden.
```
```suggestion
    /// Hides the loading view.
```

truizlop commented on project/Component/nef/Models/Error.swift:
```diff
@@ -2,17 +2,32 @@
 
+/// Models a `nef` error.
 public enum Error: Swift.Error {
+    /// Describes a `Compiler` error given detailed information.
```
```suggestion
    /// Describes a `Compiler` error with detailed information.
```

truizlop commented on project/Component/nef/Models/Error.swift:
```diff
@@ -2,17 +2,32 @@
+    /// Describes a `Compiler` error given detailed information.
     case compiler(info: String = "")
+    
+    /// Describes a `Markdown` error given detailed information.
```
```suggestion
    /// Describes a `Markdown` error with detailed information.
```

truizlop commented on project/Component/nef/Models/Error.swift:
```diff
@@ -2,17 +2,32 @@
+    /// Describes a `Markdown` error given detailed information.
     case markdown(info: String = "")
+    
+    /// Describes a `Jekyll` error given detailed information.
```
```suggestion
    /// Describes a `Jekyll` error with detailed information.
```

truizlop commented on project/Component/nef/Models/Error.swift:
```diff
@@ -2,17 +2,32 @@
+    /// Describes a `Jekyll` error given detailed information.
     case jekyll(info: String = "")
+    
+    /// Describes a `Carbon` error given detailed information.
```
```suggestion
    /// Describes a `Carbon` error with detailed information.
```

truizlop commented on project/Component/nef/Models/Error.swift:
```diff
@@ -2,17 +2,32 @@
+    /// Describes a `Carbon` error given detailed information.
     case carbon(info: String = "")
+    
+    /// Describes a `Carbon` error given detailed information about the snapshot failure.
```
```suggestion
    /// Describes a `Carbon` error with detailed information about the snapshot failure.
```

truizlop commented on project/Component/nef/Models/Error.swift:
```diff
@@ -2,17 +2,32 @@
+    /// Describes a `Carbon` error given detailed information about the snapshot failure.
     case invalidSnapshot(info: String = "")
+    
+    /// Describes a `Playground` error given detailed information.
```
```suggestion
    /// Describes a `Playground` error with detailed information.
```

truizlop commented on project/Component/nef/Models/Error.swift:
```diff
@@ -2,17 +2,32 @@
+    /// Describes a `Playground` error given detailed information.
     case playground(info: String = "")
+    
+    /// Describes a `Swift Playground` error given detailed information.
```
```suggestion
    /// Describes a `Swift Playground` error with detailed information.
```

truizlop commented on project/Component/nef/Utils/Kleisli+API.swift:
```diff
@@ -5,16 +5,29 @@ import NefModels
 import Bow
 import BowEffects
 
+/// Transforms an `EnvIO<Console, nef.Error, A>` into `IO<nef.Error, A>` using a dummy console.
```
I think this documentation is ignored.

miguelangel-dev replied:
it is optional, but if you add it, it shows as a description, on bottom of the signature:
![Captura de pantalla 2020-03-23 a las 17 19 02](https://user-images.githubusercontent.com/25568562/77338309-7956b100-6d2a-11ea-8cf9-34528fdeb14e.png)
